projects/ad7124_iio/app/app_config.c

```diff
@@ -0,0 +1,345 @@
+static struct no_os_gpio_init_param rdy_init_param = {
+	.number = RDY_PIN,
+	.port = RDY_PORT,
+	.platform_ops = &gpio_platform_ops,
+};
+
+/* GPIO - Chip select Pin init parameters */
+static struct no_os_gpio_init_param csb_init_param = {
+	.number = SPI_CSB,
+	.port = SPI_CS_PORT,
+	.platform_ops = &gpio_platform_ops,
+	.extra = NULL
+};
+
+/* External interrupt init parameters */
+static struct no_os_irq_init_param trigger_gpio_irq_params = {
+	.irq_ctrl_id = IRQ_INT_ID,
+	.platform_ops = &irq_platform_ops,
+	.extra = &ext_int_extra_init_params
+};
+
+/* External interrupt callback descriptor */
+static struct no_os_callback_desc ext_int_callback_desc = {
+	.callback = data_capture_callback,
+	.event = NO_OS_EVT_GPIO,
+	.peripheral = NO_OS_GPIO_IRQ,
+};
+
+/* Ticker interrupt descriptor */
+struct no_os_irq_ctrl_desc *ticker_int_desc;
+
+/******************************************************************************/
+/************************** Functions Declaration *****************************/
+/******************************************************************************/
+
+/******************************************************************************/
+/************************** Functions Definition ******************************/
+/******************************************************************************/
+
+#if (ACTIVE_IIO_CLIENT == IIO_CLIENT_LOCAL)
+/**
+ * @brief 	lvgl tick update callback function for pocket lab
+ * @param	ctx[in] - callback context
+ * @return	none
+ */
+void lvgl_tick_callback(void* ctx)
+{
+	pl_gui_lvgl_tick_update(LVGL_TICK_TIME_MS);
+}
+#endif
+
+/**
+ * @brief 	Initialize the UART peripheral
+ * @return	0 in case of success, negative error code otherwise
+ */
+static int init_uart(void)
+{
+	int ret;
+
+	ret = no_os_uart_init(&uart_desc, &uart_init_params);
+	if (ret) {
+		return ret;
+	}
+
+#if defined(CONSOLE_STDIO_PORT_AVAILABLE)
+	/* Initialize the serial link for console stdio communication */
+	ret = no_os_uart_init(&uart_console_stdio_desc,
+			      &uart_console_stdio_init_params);
+	if (ret) {
+		return ret;
+	}
+#endif
+
+	return 0;
+}
+
+/**
+ * @brief Initialize the IRQ contoller
+ * @return 0 in case of success, negative error code otherwise
+ * @details This function initialize the interrupts for system peripherals
+ */
+int init_interrupt(void)
+{
+	int ret;
+
+	/* Init interrupt controller for external interrupt */
+	ret = no_os_irq_ctrl_init(&trigger_irq_desc, &trigger_gpio_irq_params);
+	if (ret) {
+		return ret;
+	}
+
+#if (DATA_CAPTURE_MODE == BURST_DATA_CAPTURE)
+	ret = no_os_irq_register_callback(trigger_irq_desc,
+					  IRQ_INT_ID,
+					  &ext_int_callback_desc);
+	if (ret) {
+		return ret;
+	}
+
+	ret = no_os_irq_trigger_level_set(trigger_irq_desc,
+					  IRQ_INT_ID,
+					  NO_OS_IRQ_EDGE_FALLING);
+	if (ret) {
+		return ret;
+	}
+#endif
+
+	return 0;
+}
+
+#if (ACTIVE_PLATFORM == MBED_PLATFORM)
+/**
+ * @brief 	Initialize the lvgl timer peripheral
```

Comment:
nit: Can we format the extra space in the doxygen descriptions of all the functions? This was seen in other files too.